index.js

```diff
@@ -1 +1,319 @@
-console.log('hello world');
+import express, { response } from 'express';
+import pg from 'pg';
+import cookieParser from 'cookie-parser';
+import methodOverride from 'method-override';
+
+const { Pool } = pg;
+
+const pgConnectionConfigs = {
+  user: 'jyotikattani',
+  host: 'localhost',
+  database: 'birding',
+  port: 5432,
+};
+
+const pool = new Pool(pgConnectionConfigs);
+
+const app = express();
+app.set('view engine', 'ejs');
+app.use(express.static('public'));
+app.use('/favicon.ico', express.static('images/flamingo.png'));
+app.use(express.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(methodOverride('_method'));
+
+function createNewNote(req, res) {
+  const speciesQuery = 'SELECT * FROM species';
+  pool.query(speciesQuery, (err, result) => {
+    if (err) {
+      console.log(err);
+      return err;
+    }
+    const speciesObjArr = result.rows;
+
+    res.render('note', { speciesObjArr });
+  });
+}
+
+function handleNotePostReq(req, res) {
+  const formData = req.body;
+  console.log(formData);
+  console.log(typeof formData.date);
+  console.log(`${formData.date},${formData.time},${formData.day},${formData.flock_size},${formData.habitat}`);
+  const { date } = formData;
+  const { time } = formData;
+  const { day } = formData;
+  const flockSize = formData.flock;
+  console.log(`flock size ${flockSize}`);
+  const { habitat } = formData;
+  const { speciesId } = formData;
+
+  const { behaviour } = formData;
+
+  const notesInputData = `INSERT INTO notes (date, time, day, flock_size, habitat, species_id, behaviour) VALUES ('${date}', '${time}', '${day}', ${flockSize} , '${habitat}', '${speciesId}', '${behaviour}')`;
+
+  pool.query(notesInputData, (err, result) => {
+    if (err) {
+      console.log(err);
+      return err;
+    }
+    console.log(result);
+    // res.send(result);
+    res.redirect('/');
+  });
+}
+
+function renderSingleNote(req, res) {
+  console.log('render single note is running');
+  const noteIndex = req.params.id;
+
+  const singleNoteDate = `SELECT * FROM notes WHERE id = '${noteIndex}'`;
+  pool.query(singleNoteDate, (err, result) => {
+    if (err) {
+      return err;
+    }
+
+    const singleNote = result.rows[0];
+    // join notes and comments table.
+    const joinQuery = `SELECT * FROM comments JOIN notes ON notes.id = comments.note_id WHERE note_id = ${noteIndex}`;
+    pool.query(joinQuery, (err, joinResult) => {
+      if (err) {
```

Comment:
If no notes exist, is the error handled here? 
For example if `joinResult` doesn't result in any object?